### PR TITLE
Get mypy working a bit better

### LINKED ISSUE #305: `mypy` and `ruff` do not detect as many errors when run through `pre-commit`
### Definition of Done / Acceptance Criteria

- [x] Mypy will detect common typing errors to an appropriate level as determined by the dev team (it is highly configurable)
- [x] Ditto ruff
- [x] Satisfactorily investigate why settings are different when run directly vs via pre-commit

### Testing

_No response_

### Documentation

_No response_

### Dependencies

_No response_

### Details and Comments

I don't ever recall `mypy` failing in our pre-commit checks - is it just turned down to a very low level? Even the most obvious of typing errors isn't detected:
Quick test: add line `upload_parquet_files(42)` to `pixl_core/src/core/upload.py`.
Running `mypy pixl_core/src/core/upload.py` gives `error: Argument 1 to "upload_parquet_files" has incompatible type "int"; expected "ParquetExport"`. My IDE also gives the error.
But running `pre-commit run --all-files --verbose` passes just fine. I would expect it to fail too.

### PR DESCRIPTION
**Pull request by jeremyestein**
Possibly fixes #305 

I wish I understood why this worked, but at least mypy is giving reasonable looking errors with it now.

I'll leave this PR as a starting point for the next person who picks this up - I assume we want to fix all the errors as part of the fix? (Remember to merge/rebase first though)

According to https://github.com/pre-commit/mirrors-mypy/blob/main/.pre-commit-hooks.yaml the default value is `entry: mypy`. However, running the command line `mypy` manually does show some errors, so this is not the whole story!

`pre-commit` does run commands inside its own environment so it's possible that has an effect.

So, to summarise:
| command | result |
| --- | --- |
| `mypy` | Found 13 errors in 6 files (checked 71 source files) |
| `mypy .` | Found 67 errors in 23 files (checked 66 source files) |
| (without "entry" fix) `pre-commit run --all-files --verbose ` | Success: no issues found in 71 source files |
| (with "entry" fix) - `pre-commit run --all-files --verbose ` | Found 59 errors in 18 files (checked 66 source files) |

Make that make sense!

(`mypy.ini` is being read in all cases - put in a bad config option and it'll warn you)

### CODE REVIEW

**Comment by milanmlft** [comment]
So confusing 🙈

**Comment by milanmlft** [comment]
Fixed all remaining `mypy` errors.

File: pixl_ehr/src/pixl_ehr/_processing.py [hunk 1]
```diff
@@ -75,7 +72,7 @@ class PatientEHRData:
     procedure_occurrence_id: int
     project_name: str
     extract_datetime: datetime
-    acquisition_datetime: Optional[datetime]
+    acquisition_datetime: Optional[date]
```
Comment (stefpiatek) [inline pixl_ehr/src/pixl_ehr/_processing.py:75]: can we change the field name as well?

File: test/system_test.py [hunk 2]
```diff
@@ -62,16 +71,16 @@ def test_ftps_parquet_upload(self):
...
-        zip_files = []
+        zip_files: list[str] = []
 
         def zip_file_list() -> str:
             return f"zip files found: {zip_files}"
 
         def two_zip_files_present() -> bool:
             nonlocal zip_files
-            zip_files = list(TestFtpsUpload.expected_output_dir.glob("*.zip"))
+            zip_files = list(str(TestFtpsUpload.expected_output_dir.glob("*.zip")))
```
Comment (stefpiatek) [inline test/system_test.py:83]:
```suggestion
            zip_files = [str(x) for x in TestFtpsUpload.expected_output_dir.glob("*.zip"))]
```

**Review by stefpiatek — APPROVED** [review]
Looks good, thanks!
